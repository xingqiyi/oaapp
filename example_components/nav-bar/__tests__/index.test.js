import React from 'react';
import { render, mount } from 'enzyme';
import { renderToJson } from 'enzyme-to-json';
import NavBar from '../index';

describe('NavBar', () => {
  it('renders correctly', () => {
    const wrapper = render(
      <NavBar leftContent="返回" mode="light" rightContent={'i'}>NavBar</NavBar>,
    );
    expect(renderToJson(wrapper)).toMatchSnapshot();
  });

  it('trigger event correctly', () => {
    const onLeftClick = (e) => {
      // console.log('onLeftClick');
      expect(e.target.className).toBe('am-navbar-left');
    };
    const wrapper = mount(
      <NavBar onLeftClick={onLeftClick}>NavBar</NavBar>,
    );
    wrapper.find('.am-navbar-left').simulate('click');
  });
});
