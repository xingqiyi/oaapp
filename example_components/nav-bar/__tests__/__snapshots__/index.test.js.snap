// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavBar renders correctly 1`] = `
<div
  class="am-navbar am-navbar-light"
>
  <div
    class="am-navbar-left"
    role="button"
  >
    <span
      aria-hidden="true"
      class="am-navbar-left-icon"
    >
      <svg
        class="am-icon am-icon-left am-icon-md"
      >
        <use
          xlink:href="#left"
        />
      </svg>
    </span>
    <span
      class="am-navbar-left-content"
    >
      返回
    </span>
  </div>
  <div
    class="am-navbar-title"
  >
    NavBar
  </div>
  <div
    class="am-navbar-right"
  >
    i
  </div>
</div>
`;
